crates/ruma-macros/src/events/event_enum.rs

```diff
@@ -467,16 +467,17 @@ fn expand_accessor_methods(
         has_field(kind, var).then(|| {
             let docs = format!("Returns this event's `{}` field.", name);
             let ident = Ident::new(name, Span::call_site());
-            let field_type = field_return_type(name, ruma_common);
+            let (field_type, is_ref) = field_return_type(name, ruma_common);
             let variants = variants.iter().map(|v| v.match_arm(quote! { Self }));
             let call_parens = maybe_redacted.then(|| quote! { () });
+            let prefix = if is_ref { quote!(&) } else { quote!() };
```

Comment:
Instead of returning an extra bool from `field_return_type`, I think it's fine to just check `name == "origin_server_ts"` right now. Also I would suggest calling this `ampersand`, should be a little clearer.

Reply:
I have made the changes. Please review.